Core/src/org/sleuthkit/autopsy/contentviewers/SQLiteViewer.java

```diff
@@ -363,33 +410,103 @@ protected Integer doInBackground() throws Exception {
-
+        };
+        worker.execute();
     }
 
-    enum SQLStorageClass {
-        NULL,
-        INTEGER,
-        REAL,
-        TEXT,
-        BLOB
-    };
+    private void readTable(String tableName, int startRow, int numRowsToRead) {
 
-    private class SQLColDef {
+        if (worker != null && !worker.isDone()) {
+            worker.cancel(false);
+            worker = null;
+        }
 
-        private final String colName;
-        private final SQLStorageClass storageClass;
+        worker = new SwingWorker<ArrayList<Map<String, Object>>, Void>() {
+            @Override
+            protected ArrayList<Map<String, Object>> doInBackground() throws Exception {
+                try {
+                    Statement statement = connection.createStatement();
+                    ResultSet resultSet = statement.executeQuery(
+                            "SELECT * FROM " + tableName
+                            + " LIMIT " + Integer.toString(numRowsToRead)
+                            + " OFFSET " + Integer.toString(startRow - 1)
+                    ); //NON-NLS
 
-        SQLColDef(String colName, SQLStorageClass sc) {
-            this.colName = colName;
-            this.storageClass = sc;
+                    return resultSetToArrayList(resultSet);
+                } catch (SQLException ex) {
+                    LOGGER.log(Level.SEVERE, "Failed to get data for table.", ex); //NON-NLS
+                }
+                //NON-NLS
+                return null;
+            }
+
+            @Override
+            protected void done() {
+
+                if (isCancelled()) {
+                    return;
+                }
+
+                super.done();
+                try {
+                    ArrayList<Map<String, Object>> rows = get();
+                    if (Objects.nonNull(rows)) {
+                        selectedTableView.setupTable(rows);
+                    }else{
+                        selectedTableView.setupTable(Collections.emptyList());
+                    }
+                } catch (InterruptedException | ExecutionException ex) {
+                    LOGGER.log(Level.SEVERE, "Unexpected exception while reading table.", ex); //NON-NLS
+                }
+            }
+        };
+
+        worker.execute();
+    }
+
+    private ArrayList<Map<String, Object>> resultSetToArrayList(ResultSet rs) throws SQLException {
+        ResultSetMetaData md = rs.getMetaData();
+        int columns = md.getColumnCount();
+        ArrayList<Map<String, Object>> arraylist = new ArrayList<>();
+        while (rs.next()) {
+            Map<String, Object> row = new LinkedHashMap<>(columns);
+            for (int i = 1; i <= columns; ++i) {
+                if (rs.getObject(i) == null) {
+                    row.put(md.getColumnName(i), "");
+                } else {
+                    if (md.getColumnTypeName(i).compareToIgnoreCase("blob") == 0) {
+                        row.put(md.getColumnName(i), "BLOB Data not shown...");
```

Comment:
It would be better if the message about the BLOB data was implemented via NbBundle.Messages instead of hard-coded.  This will make it easier/less expensive for localization. Also, I would make the message just "BLOB data not shown", lower-casing "data" and omitting the ellipsis which is generally used when the user can invoke a dialog. I personally find the use of the ellipsis slightly unusual here.